src/app/shared/components/routed-dialog-wrapper/routed-dialog-wrapper.component.ts

```diff
@@ -97,4 +66,24 @@ export class RoutedDialogWrapperComponent
     this.closedByNavigate = true;
     unsubscribeAll(this.subscriptions);
   }
+
+  private openDialog(component: any, dialogData: any, dialogOptions: any): void {
+    const config = {
+      ...dialogOptions,
+      data: dialogData,
+      hasBackdrop: true,
+      backdropClass: 'cdk-overlay-dark-backdrop',
+      panelClass: 'custom-dialog-panel', // Add any custom class you want
+    };
+
+    this.dialogRef = this.dialogService.open(component, config);
+
+    this.dialogRef.closed.subscribe(() => {
+      this.dialogRef = null;
+      if (!this.closedByNavigate) {
+        const targetRoute = this.routeHistory.previousRoute || '/app/';
+        this.router.navigateByUrl(targetRoute);
+      }
```

Comment:
While sometimes these codecov warnings can be ignored, this is a good example of a place where test coverage would have been useful as part of the refactoring process. If I were restructuring this specific bit of code, I would have written tests to cover the original routed dialog wrapper *before* doing any CDK Dialog conversions. Once these were written and passed, *then* I would begin refactoring. In an ideal world, once we restructure the RoutedDialogWrapper the tests would still pass proving that our refactoring didn't break anything. In practice, we'd probably have to edit the tests to match the new API of the CDK Dialog. But I do recommend writing some tests up front before doing actual restructuring, so that they [characterize  the preexisting behavior](https://en.wikipedia.org/wiki/Characterization_test) and we can verify that behavior hasn't changed.